modal/_traceback.py

```diff
@@ -74,11 +74,15 @@ def append_modal_tb(exc: BaseException, tb_dict: TBDictType, line_cache: LineCac
 
 def reduce_traceback_to_user_code(tb: Optional[TracebackType], user_source: str) -> TracebackType:
```

Comment:
side note but this code seems a bit hacky and i wonder if it would make sense to not serialize traceback objects but rather have utilities for translating back and forth to an internal protobuf-based format. we would need that to support other languages in the future anyway.

completely unrelated to this PR, just something i've thought about in the past

Reply:
I don't think this function has much to do with the need to serialize tracebacks. We have it because execution running through a lot of modal framework (the cli, all the internal stuff with nested with closures getting passed around etc., and then also synchronicity) means that raw tracebacks have a _lot_ of frames that aren't relevant to a user who wants to debug their code.